### PR TITLE
fix: Support setting global tags from labels, annotations, etc (#44)

### LINKED ISSUE #44: Support setting global tags from labels, annotations, etc
I'd like to be able to set global tags based on annotations/labels/k8s namespace/etc. More specifically, I'd like to define a class like this:

```yaml
classes:
  data:
    foo: |+
      [global_tags]
        hostname = "$HOSTNAME"
        nodename = "$NODENAME"
        app = "$KUBERNETES_LABEL_APP"
```

To be clear, obviously I can _already_ define a class like that, but it won't work because there's no way to get the environment variable `$KUBERNETES_LABEL_APP` populated in the Telegraf container. If I was creating the container myself I'd be able to do something like,
```
env:
  - name: KUBERNETES_LABEL_APP
    valueFrom:
      fieldRef:
        fieldPath: metadata.labels['app']
```
But there's no way to set up anything like that when the container is injected automatically by the Operator. 

As an alternative, being able to just specify `telegraf.influxdata.com/include-labels: 'app,whatever'` or `telegraf.influxdata.com/tags: 'app=foo,whatever=bar'` as an annotation and have the Operator automatically construct...

```toml
[global_tags]
app = "foo"
whatever = "bar"
```
...would also be pretty good, but I get the feeling that could get awkward to implement since you can't define the same table multiple times in a TOML document, so if the class defines a `[global_tags]` table as well then they'd clash.

### PR DESCRIPTION
**Pull request by cbos**
Closes #44

I made the implementation as discussed on #44 

- [x] Signed the [InfluxData CLA](https://www.influxdata.com/legal/cla/)
  ([if necessary](https://github.com/influxdata/telegraf-operator/blob/master/CONTRIBUTING.md#sign-the-influxdata-cla))
- [x] Tests pass (no build errors)
- [x] Rebased/mergeable

### CODE REVIEW

**Review by wojciechka — COMMENTED** [review]
Looks good.

There is one thing I believe should be fixed - `len() == 2` check.

The variable name including `Prefix` is just my preference and I am fine with it being either way.

Once the length check is in place, I'm more than happy to approve it and see it merged so we can release a new version.

**Comment by cbos** [comment]
@wojciechka I made the required changes.
Without the len check it indeed crashed, the testcase got a panic.
I also added the Prefix keyword to the variables, that indeed makes it more clear.

File: sidecar.go [hunk 1]
```diff
@@ -417,9 +425,75 @@ func (h *sidecarHandler) newContainer(pod *corev1.Pod, containerName string) (co
...
+				ValueFrom: &corev1.EnvVarSource{
+					ConfigMapKeyRef: &corev1.ConfigMapKeySelector{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: selector[0],
+						},
+						Key: selector[1],
+					},
+				},
+			})
+		}
```
Comment (wojciechka) [inline sidecar.go:464]:
```suggestion
		} else {
			h.Logger.Info("unable to parse configmapkeyref %s with value of \"%s\"", name, value)
		}
```

File: sidecar.go [hunk 2]
```diff
@@ -417,9 +425,75 @@ func (h *sidecarHandler) newContainer(pod *corev1.Pod, containerName string) (co
...
+				ValueFrom: &corev1.EnvVarSource{
+					SecretKeyRef: &corev1.SecretKeySelector{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: selector[0],
+						},
+						Key: selector[1],
+					},
+				},
+			})
+		}
```
Comment (wojciechka) [inline sidecar.go:482]:
```suggestion
		} else {
			h.Logger.Info("unable to parse secretkeyref %s with value of \"%s\"", name, value)
		}
```

**Review by wojciechka — APPROVED** [review]
I think it looks good now, but you could also log the problem so it is easier to troubleshoot.

I have added a suggestion, but I don't think that's blocking this change from being merged.

Please decide if you like the suggested change and merge the code :)

**Review by cbos — COMMENTED** [review]
I applied the suggestions. It is indeed good to provide the feedback.

**Comment by cbos** [comment]
Can you merge it? I don have the rights to do so, I can only create a PR, I don't have write permissions.

**Pull request merged**